docker/tools/visualizer/Dockerfile.visualizer

```
@@ -0,0 +1,87 @@
+# cspell:ignore openbox, VNC, tigervnc, novnc, websockify, newkey, xstartup, keyout
+
+### Builder
+FROM ghcr.io/autowarefoundation/autoware:universe-visualization-devel AS builder
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+ENV CCACHE_DIR="/root/.ccache"
+ARG ROS_DISTRO
+WORKDIR /autoware
+COPY src/simulator /autoware/src/simulator
+COPY docker/scripts/resolve_rosdep_keys.sh /autoware/resolve_rosdep_keys.sh
+RUN chmod +x /autoware/resolve_rosdep_keys.sh
+
+# Build simulator messages and rviz plugins ONLY
+# hadolint ignore=SC1091
+RUN --mount=type=ssh \
+  --mount=type=cache,target=/var/cache/apt,sharing=locked \
+  apt-get update \
+  && source /opt/ros/"$ROS_DISTRO"/setup.bash && source /opt/autoware/setup.bash \
+  && rosdep update && rosdep install -y --from-paths src --ignore-src --rosdistro $ROS_DISTRO \
+  && colcon build --cmake-args \
+    "-Wno-dev" \
+    "--no-warn-unused-cli" \
+    --install-base /opt/autoware \
+    --merge-install \
+    --mixin release compile-commands ccache \
+    --base-paths /autoware/src \
+    --packages-up-to-regex ".*_msgs$" ".*rviz_plugin$" \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/* "$HOME"/.cache
+
+# Extract rosdep dependencies for visualizer
+# hadolint ignore=SC1091
+RUN source /opt/ros/"$ROS_DISTRO"/setup.bash && source /opt/autoware/setup.bash \
+    && /autoware/resolve_rosdep_keys.sh /autoware/src ${ROS_DISTRO} \
```

I think this PR needs only `exec_depend` dependencies.

```suggestion
    && /autoware/resolve_rosdep_keys.sh /autoware/src ${ROS_DISTRO} --dependency-types=exec \
```